Use workgroupMappingDim in rocroller_host

marbre opened the pull request:
The API for command solution changed so use the `workgroupMappingDim` rather than `workgroupMapping`.

davidd-amd committed:
Use workgroupMappingDim in rocroller_host

The api for command solution changed so
use the workgroupMappingDim rather than workgroupMapping

davidd-amd reviewed (commented):
This won't work until we start pulling rocroller via add_subdirectory... Or in this PR we can bump rocroller and in a follow up PR we move to rocroller. The latter approach is more incremental.

davidd-amd committed:
Update rocroller hash

davidd-amd committed:
Changes to support bumping rocroller

davidd-amd committed:
Do not leak boost header target

The rocroller target links Boost::headers which may cause configure
errors for some versions of boost where it is not exported. It was
made public because all of the rocroller headers are unnecessarily
public. Making it private did not break builds which suggests that it
can be private for the time being.

davidd-amd committed:
Additional guard against leaking target

awhittle3 commented:
Request closure of this PR: Work for this has been achieved in PR #1588.

memmett commented:
Closing this.  Please reopen if desired.
